lib/SIL/TypeLowering.cpp

```diff
@@ -2139,9 +2140,22 @@ TypeConverter::getLoweredLocalCaptures(AnyFunctionRef fn) {
 
         // Fall through to capture the storage.
       }
-      
+
       // Collect non-function captures.
-      captures.insert(capture);
+      ValueDecl *value = capture.getDecl();
+      if (capturedValues.count(value)) {
+        for (auto it = captures.begin(); it != captures.end(); ++it) {
+          if (it->getDecl() == value) {
+            // The value is already in the list, it needs to have the logical
+            // AND of each flag of all uses.
+            *it = CapturedValue(value, it->getFlags() & capture.getFlags());
```

Comment:
This seems very subtle. Can you add a `mergeFlags` method to CapturedValue or something, so that if we add a third flag that doesn't compose this way we don't have to track it down later?

Reply:
Will do!